fabric-common-tags-api-v1/src/main/java/net/fabricmc/fabric/impl/common/tag/datagen/generators/ItemTagGenerator.java

```diff
@@ -0,0 +1,127 @@
+package net.fabricmc.fabric.impl.common.tag.datagen.generators;
+
+import net.minecraft.item.Items;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.api.datagen.v1.FabricDataGenerator;
+import net.fabricmc.fabric.api.datagen.v1.provider.FabricTagProvider;
+import net.fabricmc.fabric.api.tags.v1.CommonItemTags;
+
+public class ItemTagGenerator extends FabricTagProvider.ItemTagProvider {
+	/**
+	 * @deprecated Use {@link CommonItemTags#PICKAXES}
+	 */
+	@Deprecated
+	private static final Identifier FABRIC_PICKAXES = createFabricId("pickaxes");
+	/**
+	 * @deprecated Use {@link CommonItemTags#SHOVELS}
+	 */
+	@Deprecated
+	private static final Identifier FABRIC_SHOVELS = createFabricId("shovels");
+	/**
+	 * @deprecated Use {@link CommonItemTags#HOES}
+	 */
+	@Deprecated
+	private static final Identifier FABRIC_HOES = createFabricId("hoes");
+	/**
+	 * @deprecated Use {@link CommonItemTags#AXES}
+	 */
+	@Deprecated
+	private static final Identifier FABRIC_AXES = createFabricId("axes");
+	/**
+	 * @deprecated Use {@link CommonItemTags#SHEARS}
+	 */
+	@Deprecated
+	private static final Identifier FABRIC_SHEARS = createFabricId("shears");
+	/**
+	 * @deprecated Use {@link CommonItemTags#SWORDS}
+	 */
+	@Deprecated
+	private static final Identifier FABRIC_SWORDS = createFabricId("swords");
+	
+	public ItemTagGenerator(FabricDataGenerator dataGenerator) {
+		super(dataGenerator);
+	}
+
+	@Override
+	protected void generateTags() {
+		generateToolTags();
+		generateExcludedItemTag();
+	}
+
+	private void generateExcludedItemTag() {
+		getOrCreateTagBuilder(CommonItemTags.ITEM_INVENTORY_EXCLUDED)
```

Comment:
I am trying to understand what this means, but I honestly don't understand.

Reply:
It is a method where the population of the tag entries for that tag occur

Reply:
I am more talking about the tag itself; what does it mean?

Reply:
It was a suggestion on fabricord, see https://github.com/FabricMC/fabric/pull/2063/files/532c1a9bb07d2ddfbcba76fdf2d9ba7846190385#r825227686

The idea was a tag containing items that are rejected from shulker boxes or other item inventories.

I am unsure if it will be included as more discussion is needed.